SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialBasicAwaiter.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/SpatialBasicAwaiter.h"
+
+#include "Engine/World.h"
+#include "TimerManager.h"
+
+const FString USpatialBasicAwaiter::TIMEOUT_MESSAGE = FString{ TEXT("Timed out while waiting to become Ready!") };
+constexpr TCHAR DESTROYED_MESSAGE_TEMPLATE[] = TEXT("Awaiter '%s' was destroyed!");
+
+void USpatialBasicAwaiter::BeginDestroy()
+{
+	InvokeQueuedDelegates(FString::Printf(DESTROYED_MESSAGE_TEMPLATE, *GetName()));
+	Super::BeginDestroy();
+}
+
+FDelegateHandle USpatialBasicAwaiter::Await(const FOnReady& OnReadyDelegate, const float Timeout)
+{
+	if (bIsReady)
```

Comment:
Could you decorate a couple of comments here about what each path means.
ie. it's already ready, so no need to wait, execute immediately